src/overlays/actors/ovl_En_Hata/z_en_hata.c

```diff
@@ -27,15 +26,106 @@ const ActorInit En_Hata_InitVars = {
     (ActorFunc)EnHata_Update,
     (ActorFunc)EnHata_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hata/EnHata_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hata/EnHata_Destroy.s")
+static UNK_TYPE4 EnHata_UnusedData[] = { 0x0A000939, 0x20010000, 0x00000000, 0x00000000, 0x00000000,
+                                         0x00000080, 0x00000000, 0x00050100, 0x001000F6, 0x00000000,
+                                         0x00000000, 0x00000000, 0x00000000, 0xFF000000 };
+
+static Vec3f EnHata_Vec = { 0, 0, 0 };
+
+extern AnimationHeader D_06000444;
+extern SkeletonHeader D_06002FD0;
+extern u32 D_060000C0;
+
+void EnHata_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHata* this = THIS;
+    SkelAnime* skelAnime;
+    void* collisionPtr;
```

Comment:
I believe you based this name on the function argument for that function but it's actually wrong. It should get fixed eventually as z_bgcheck.c gets worked on.
```suggestion
    u32 temp;
```